rules/xcodeproj.bzl

```diff
@@ -180,28 +200,44 @@ def _xcodeproj_impl(ctx):
             "optional": True,
             "buildPhase": "none",
         } for s in target_info.asset_srcs.to_list()]
+        target_settings = {
+            "PRODUCT_NAME": target_info.name,
+            "BAZEL_BIN_SUBDIR": target_info.bazel_bin_subdir,
+            "MACH_O_TYPE": target_macho_type,
+        }
+
+        if target_info.product_type == "application":
+            target_settings["INFOPLIST_FILE"] = "$BAZEL_STUBS_DIR/Info-stub.plist"
+            target_settings["PRODUCT_BUNDLE_IDENTIFIER"] = target_info.bundle_id
+        if target_info.product_type == "bundle.unit-test":
+            target_settings["SUPPORTS_MACCATALYST"] = False
```

Comment:
For some reason, when running tests on using legacy build system, I had to set this to false in xcode build settings.